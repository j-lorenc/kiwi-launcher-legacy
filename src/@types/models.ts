export class Game {
  id: string;
  name: string;
  originalId: string;
  iconUrl?: string;
  backgroundUrl?: string;
  coverUrl?: string;
  lastPlayed?: number;
  playtime?: number;
  installed?: boolean;
  verticalUrl?: string;
  logo?: string;

  constructor(
    id: string,
    name: string,
    originalId: string,
    iconUrl?: string,
    backgroundUrl?: string,
    coverUrl?: string
  ) {
    this.id = id;
    this.name = name;
    this.originalId = originalId;
    this.iconUrl = iconUrl;
    this.backgroundUrl = backgroundUrl;
    this.coverUrl = coverUrl;
    if (coverUrl) {
      this.verticalUrl = `file://C:\\Program Files (x86)\\Steam\\appcache\\librarycache\\${originalId}_library_600x900.jpg`;
    }
    if (coverUrl) {
      this.logo = `file://C:\\Program Files (x86)\\Steam\\appcache\\librarycache\\${originalId}_logo.png`;
    }
  }
}
