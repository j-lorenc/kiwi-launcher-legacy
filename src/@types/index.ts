import { Dispatch } from 'react';

export interface GameFilter {
  gameName: string;
}

export type FilterAction = { type: 'filter'; payload: string };

export interface FilterContextType {
  state: GameFilter;
  dispatch: Dispatch<FilterAction>;
}
