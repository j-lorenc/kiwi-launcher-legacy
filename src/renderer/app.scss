@import './stylesheets/reset';

h1 {
  margin: 0;
}

#root {
  height: 100vh;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  background-color: rgb(0, 0, 0);
}

button {
  border: none;
  background: none;

  &:focus,
  &:active {
    border: none;
    outline: none;
  }
}

input {
  font-family: 'SairaExtraCondensed';
  font-size: 14px;

  background: rgba(0, 0, 0, 0.5);
  color: white;
  border-right: 1px solid rgba(255, 255, 255, 0.17);
  border-bottom: 1px solid rgba(255, 255, 255, 0.17);
  border-top: 1px solid rgba(0, 0, 0, 0.2);
  border-left: 1px solid rgba(0, 0, 0, 0.2);

  letter-spacing: 0.5px;

  outline: none;
  border-radius: 3px;
  transition: 0.15s width;

  &:focus,
  &:active,
  &.active {
    border-right: 1px solid rgba(181, 230, 29, 1);
    border-bottom: 1px solid rgba(181, 230, 29, 1);

    border-top: 1px solid rgba(180, 230, 29, 0.6);
    border-left: 1px solid rgba(181, 230, 29, 0.6);

    transition: 0.3s width;
  }
}

/* width */
::-webkit-scrollbar {
  width: 12px;
}

/* Track */
::-webkit-scrollbar-track {
  width: 15px;
  background: rgba(0, 0, 0, 0);
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: rgba(40, 40, 40, 0.75);
  border-radius: 1px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: rgba(70, 70, 70, 0.75);
}
