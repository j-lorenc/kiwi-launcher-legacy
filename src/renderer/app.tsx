import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { AppContainer } from 'react-hot-loader';

import Application from './components/Application';

const render = (Component: () => JSX.Element) => {
  ReactDOM.render(
    <AppContainer>
      <Component />
    </AppContainer>,
    document.getElementById('root')
  );
};

render(Application);
