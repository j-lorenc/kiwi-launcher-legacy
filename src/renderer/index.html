<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta
      http-equiv="Content-Security-Policy"
      content="default-src 'self'; frame-src https://www.youtube.com/; style-src 'unsafe-inline' 'self'; img-src 'self' file://*  https://* http://media.steampowered.com/; child-src 'none'; script-src inline 'self' 'unsafe-inline' 'unsafe-eval' http://localhost:2003; connect-src ws://localhost:2003 http://localhost:2003 https://api.steampowered.com/ https://embed.gog.com/ https://api.gog.com/ https://auth.gog.com/ https://menu.gog.com/ https://store.steampowered.com/ https://*.igdb.com/ https://graphql.epicgames.com/ https://store-content.ak.epicgames.com"
    />
    <meta name="theme-color" content="#000000" />
    <title>Kiwi Launcher</title>
  </head>
  <body>
    <div id="root"></div>
    <div id="modal-container"></div>
  </body>
</html>
