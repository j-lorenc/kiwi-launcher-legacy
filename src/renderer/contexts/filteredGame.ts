import React, { useContext } from 'react';
import { GameFilter, FilterAction, FilterContextType } from '../../@types';

export const filterReducer = (state: GameFilter, action: FilterAction): GameFilter => {
  switch (action.type) {
    case 'filter': {
      return {
        ...state,
        gameName: action.payload,
      };
    }
  }
};

export const initialFilter: GameFilter = {
  gameName: '',
};

const FilterContext = React.createContext<FilterContextType>({
  state: initialFilter,
  dispatch: () => null,
});

export const useFilterContext = (): FilterContextType => {
  return useContext<FilterContextType>(FilterContext);
};

export default FilterContext;
