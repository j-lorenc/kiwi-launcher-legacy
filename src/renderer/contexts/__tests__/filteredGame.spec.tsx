import React from 'react';

import renderer from 'react-test-renderer';

import { filterReducer, useFilterContext } from '../filteredGame';
import { FilterAction } from '../../../@types';

describe('filterReducer', () => {
  it('should set state correctly when action type is filter', () => {
    const initialFilter = {
      gameName: '',
    };

    const action: FilterAction = {
      type: 'filter',
      payload: 'testGame',
    };

    const newState = filterReducer(initialFilter, action);

    expect(newState.gameName).toEqual('testGame');
  });

  it('should set initial context correctly', () => {
    const TestComponent = () => {
      const { state } = useFilterContext();

      return <div>{state.gameName}</div>;
    };

    const tree = renderer.create(<TestComponent />);
    expect(tree).toMatchSnapshot();
  });
});
