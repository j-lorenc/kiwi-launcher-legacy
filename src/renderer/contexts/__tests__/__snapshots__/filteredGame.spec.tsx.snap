// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterReducer should set initial context correctly 1`] = `
<div>
  
</div>
`;
